calc-fee/src/lib.rs

```diff
@@ -18,48 +19,70 @@ pub struct JSCoefficient {
 	degree: u8,
 }
 
+#[derive(Debug)]
 struct Coefficient {
 	coeff_integer: Balance,
 	coeff_frac: Perbill,
 	negative: bool,
 	degree: u8,
 }
 
+#[derive(Debug)]
 enum Multiplier {
-	Current((FixedI128, bool)),
-	Legacy(Fixed128Legacy),
+	V0(Fixed128Legacy),
+	V1((FixedI128, bool)),
+	V2(FixedU128),
 }
 
 impl Multiplier {
-	fn new(inner: i128, use_legacy: bool, bug: bool) -> Self {
-		if use_legacy {
-			Self::Legacy(Fixed128Legacy::from_parts(inner))
-		} else {
-			Self::Current((FixedI128::from_inner(inner), bug))
-		}
+	fn new(inner: &str, spec_name: &str, spec_version: u32) -> Option<Self> {
+		use Multiplier::{V0, V1, V2};
+		let mult = match (spec_name, spec_version) {
+			("polkadot", 0) => V1((new_i128(inner), true)),
+			("polkadot", v) if v < 11 => V1((new_i128(inner), false)),
+			("polkadot", v) if 11 <= v => V2(new_u128(inner)),
+
+			("kusama", 1062) => V0(new_legacy_128(inner)),
+			("kusama", v) if 1062 < v && v < 2011 => V1((new_i128(inner), false)),
+			("kusama", v) if 2011 <= v => V2(new_u128(inner)),
+
+			("westend", 10) => V0(new_legacy_128(inner)),
+			("westend", v) if 10 < v && v < 31 => V1((new_i128(inner), false)),
+			("westend", v) if 31 <= v => V2(new_u128(inner)),
+
+			_ => {
+				info!("Unsupported runtime: {}#{}", spec_name, spec_version);
+				return None;
```

Comment:
This warning is a good idea, and I think it works well for now. It does make me think though that in the future we should maybe think about how to make _all_ of sidecar easily configurable for a wide array of substrate based chains. Polkadot-js already easily supports adding types, and I am wondering if we could do something of similar effect with fee stuff in this crate.

Reply:
Instead of bailing we could just use the current `Multiplier` as long as the chain supplied a valid `Polynomial`. That would instantly support all current chains that use the `ChargeTransactionPayment` (should be most of them).

So maybe we should rather explicitly ban unsupported versions of our runtimes than disallowing anything we do not know.